sanic/cookies.py

```diff
@@ -103,6 +104,9 @@ def __setitem__(self, key, value):
         if key not in self._keys:
             raise KeyError("Unknown cookie property")
         if value is not False:
+            if key.lower() == "max-age":
+                if not str(value).isdigit():
+                    value = DEFAULT_MAX_AGE
```

Comment:
Since you already did the check here, we can get rid of the `try/except` block for `max-age` inside `encode` method.